std/experimental/color/rgb.d

```diff
@@ -0,0 +1,852 @@
+module std.experimental.color.rgb;
+
+import std.experimental.color;
+import std.experimental.color.xyz;
+
+import std.traits: isInstanceOf, isNumeric, isIntegral, isFloatingPoint, isSigned, isSomeChar, Unqual;
+import std.typetuple: TypeTuple;
+import std.typecons: tuple;
+
+@safe: pure: nothrow: @nogc:
+
+
+enum isRGB(T) = isInstanceOf!(RGB, T);
+
+enum RGBColorSpace
+{
+    sRGB,
+    sRGB_Gamma2_2,
+/+ TODO: these all need testing
```

Comment:
Of course it's *possible*, but you're arguing against the definition of a colour space.
AdobeRGB with gamma 1.8 isn't AdobeRGB anymore.
Technically, sRGB with gamma 2.2 isn't sRGB, which is why I described it as a distinct colour space.

It's possible for a user to describe an arbitrary custom colour space, the machinery is all here in this file (although I haven't really made an API that focuses on it). But it's wrong for a defined colour space to have an improper gamma ramp.

Perhaps you're conflating the common concept of user-tweakable gamma curves? This is typically a user-facing option to adjust for poor quality or improperly calibrated monitors. That concept has no business here though. Gamma ramp definitions are immutable.